gc/base/MemoryPoolLargeObjects.cpp

```diff
@@ -539,7 +539,11 @@ void*
 MM_MemoryPoolLargeObjects::findFreeEntryEndingAtAddr(MM_EnvironmentBase* env, void* addr)
 {
 	if (addr >= _currentLOABase) {
-		return _memoryPoolLargeObjects->findFreeEntryEndingAtAddr(env, addr);
+		if (getCurrentLOASize() == getApproximateFreeLOAMemorySize()) {
+			return _memoryPoolSmallObjects->findFreeEntryEndingAtAddr(env, (void*)(((uintptr_t)addr) - _loaSize));
```

Comment:
use getCurrentLOASize() instead of _loaSize here and a couple of other spots

Reply:
and try to use a single return, like you did in getAvailableContractionSizeForRangeEndingAt

Reply:
shouldn't we try to find free entry ending at _currentLOABase?